WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m

```diff
@@ -626,8 +705,10 @@ - (NSManagedObjectContext *)managedObjectContext
 - (NSFetchRequest *)fetchRequest
 {
     NSString *sortKey               = NSStringFromSelector(@selector(timestamp));
+    NSPredicate *predicate          = [NSPredicate predicateWithFormat:@"deletionInProgress == %@ || deletionInProgress == nil", @(NO)];
```

Comment:
Soooooo..... You might try something like
`[NSPredicate predicateWithFormat:@"SELF NOT IN %@", notificationIdsWithPendingDeletion]`
and not have to change the data model to support this feature. :)